agentendpoint/agentendpoint.go

```diff
@@ -104,7 +104,7 @@ func (c *Client) RegisterAgent(ctx context.Context) error {
 	}
 
 	req := &agentendpointpb.RegisterAgentRequest{AgentVersion: agentconfig.Version(), SupportedCapabilities: agentconfig.Capabilities()}
-	clog.Debugf(ctx, "Calling RegisterAgent with request:\n%s", util.PrettyFmt(req))
+	clog.DebugStructured(ctx, req, "Calling RegisterAgent with request:\n%s", util.PrettyFmt(req))
```

Comment:
I think we will want to wrap with a structure here, proabbly just move the log to after c.raw.ReceiveTaskNotification and add both body and response.

We can use a shared structure for all RPCs (proably add a common rpc logger formater to this package), so it would look something like {"Method": "RegisterAgent", "Request": {}, "Response": {}}
Make sure to redact the InstanceIdToken from the request before passing to this shared function, we dont want the function to rely on reflection.
Unsure if there are fields that would also make sense to add (grpc status maybe?), but to start just method, request, response should be enough.

Reply:
Done, with some additional work to make the proto messages digestible by cloud logging.
 
I don't understand one point from your comment - what has redacting `InstanceIdToken` to do with reflection?